libmamba/src/core/util.cpp

```diff
@@ -71,6 +71,16 @@ namespace mamba
         return status.type() != fs::file_type::not_found || status.type() == fs::file_type::symlink;
     }
 
+    bool is_writable(const fs::path& path) noexcept
+    {
+        static constexpr auto writable_flags
+            = fs::perms::owner_write | fs::perms::group_write | fs::perms::others_write;
+        std::error_code ec;
+        const auto status = fs::status(path, ec);
+        return !ec && status.type() == fs::file_type::not_found
+               && (status.permissions() & writable_flags) != fs::perms::none;
```

Comment:
Yes, but I don't know how to check that we are the owner or not without attempting to open the file/modify the directory. So I started by only checking if at least someone can write.

Reply:
I'm open to suggestion as to how to verify that, I think attempting a write is the best but also weird.

Reply:
I searched a little bit and there is a function called `access(...)`. However, the manpage also says that it creates a security hole (which I am not completely understanding if it would be relevant for us): https://linux.die.net/man/2/access

Anyways, I think this PR is already an improvement, right?

Reply:
Actually, we do have another `is_writable` function here:

https://github.com/mamba-org/mamba/blob/4ac84bf0f4624632e5bdbf99f842c1c621efaf12/libmamba/include/mamba/core/fsutil.hpp#L82-L101

Maybe better to use that one?